chat_shell/chat_shell/agents/graph_builder.py

```
@@ -83,6 +90,203 @@ def __init__(self, reason: str, has_tool_calls: bool = False):
         super().__init__(f"Tool call truncated: {reason}")
 
 
+def _env_bool(name: str, default: bool = False) -> bool:
+    """Read a boolean value from environment variables."""
+    raw = os.getenv(name)
```

Ruff (0.15.6)

[error] Undefined name `os`

(F821)